packages/dev-kit/src/rpc/wrap-rpc.ts

```diff
@@ -0,0 +1,126 @@
+import type { ChildProcess } from 'child_process';
+
+import { RpcExitError } from './rpc-error';
+import { RpcRemoteMethod, RpcMessage, RpcGMCallTypes } from './types';
+
+interface WrapRpcOptions {
+  id: string;
+  once?: boolean;
+}
+
+function createControlledPromise<T = unknown>() {
+  let resolve: (value: T) => void = () => undefined;
+  let reject: (error: unknown) => void = () => undefined;
+  const promise = new Promise<T>((aResolve, aReject) => {
+    resolve = aResolve;
+    reject = aReject;
+  });
+
+  return {
+    promise,
+    resolve,
+    reject,
+  };
+}
+
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+export function wrapRpc<T extends (...args: any[]) => any>(
+  childProcess: ChildProcess,
+  options: WrapRpcOptions,
+): RpcRemoteMethod<T> {
+  return (async (...args: unknown[]): Promise<unknown> => {
+    if (!childProcess.send) {
+      throw new Error(`Process ${childProcess.pid} doesn't have IPC channels`);
+    } else if (!childProcess.connected) {
+      throw new Error(
+        `Process ${childProcess.pid} doesn't have open IPC channels`,
+      );
+    }
+    const { id, once } = options;
+
+    // create promises
+    const {
+      promise: resultPromise,
+      resolve: resolveResult,
+      reject: rejectResult,
+    } = createControlledPromise<T>();
+    const {
+      promise: sendPromise,
+      resolve: resolveSend,
+      reject: rejectSend,
+    } = createControlledPromise<void>();
+
+    const handleMessage = (message: RpcMessage) => {
+      if (message?.id === id) {
+        if (message.type === RpcGMCallTypes.RESOLVE) {
+          // assume the contract is respected
+          resolveResult(message.value as T);
+          if (once) {
+            // declare below
+            // eslint-disable-next-line @typescript-eslint/no-use-before-define
+            removeHandlers();
+          }
+        } else if (message.type === RpcGMCallTypes.REJECT) {
+          rejectResult(message.error);
+          if (once) {
```

Comment:
there's no need to handle it in this way. If I remember correcly, childprocess already exposes a `once` function to correctly handle it.